src/main/java/com/github/manolo8/darkbot/core/manager/HangarManager.java

```diff
@@ -0,0 +1,151 @@
+                        .getAsJsonObject().get("ret").getAsJsonObject().get("hangars").getAsJsonArray();
+
+                for (JsonElement hangar : hangarArray) {
+                    if (hangar.getAsJsonObject().get("hangar_is_active").getAsBoolean()) {
+                        JsonArray dronesArray = hangar.getAsJsonObject().get("general").getAsJsonObject().get("drones").getAsJsonArray();
+                        for (JsonElement dron : dronesArray){
+                            JsonObject dronJson = dron.getAsJsonObject();
+                            String lootId = "drone_iris";
+                            switch (dronJson.get("L").getAsInt()){
+                                case 1:
+                                    lootId = "drone_flax";
+                                    break;
+                                case 2:
+                                    lootId = "drone_iris";
+                                    break;
+                                case 3:
+                                    lootId = "drone_apis";
+                                    break;
+                                case 4:
+                                    lootId = "drone_zeus";
+                                    break;
+                            }
+                            int damage = Integer.parseInt(dronJson.get("HP").getAsString().replace("%"," ").trim());
+                            this.drones.add(new Drone(lootId,dronJson.get("repair").getAsInt(),dronJson.get("I").getAsString(),
+                                    dronJson.get("currency").getAsString(),dronJson.get("LV").getAsInt(),damage));
+                        }
+                    }
+                }
+            }
+
+        } catch (Exception e){
+            e.printStackTrace();
+        }
+    }
+
+    private boolean repairDron(Drone drone, String activeHangarId){
+        try {
+            String decodeParams =
+                    "{\"action\":\"repairDrone\",\"lootId\":\"" + drone.getLootId() + "\",\"repairPrice\":" + drone.getRepairPrice() +
+                            ",\"params\":{\"hi\":" + activeHangarId + "}," +
+                            "\"itemId\":\"" + drone.getItemId() + "\",\"repairCurrency\":\"" + drone.getRepairCurrency() +
+                            "\",\"quantity\":1,\"droneLevel\":" + drone.getDroneLevel() + "}";
+            String encodeParams = Base64.getEncoder().encodeToString(decodeParams.getBytes("UTF-8"));
+            String url = "flashAPI/inventory.php?action=repairDrone&params="+encodeParams;
+            String json = this.backpageManager.getDataInventory(url);
+            if (json.contains("'isError':0")){
+                return true;
+            } else {
+                return false;
+            }
+        } catch (Exception e){
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    public void updateHangars() {
+        String params = "flashAPI/inventory.php?action=getHangarList";
```

Comment:
Same as before, we can probs put this in a constant urls file. Not too important rn